pkg/server/server.go

```diff
@@ -532,3 +541,79 @@ func goContext(parent genericapiserver.PostStartHookContext) context.Context {
 func (s *Server) WaitForPhase1Finished() {
 	<-s.rootPhase1FinishedCh
 }
+
+func (s *Server) leaderElectAndStartControllers(ctx context.Context) {
+	logger := klog.FromContext(ctx)
+
+	leaderElectionConfig := rest.CopyConfig(s.GenericConfig.LoopbackClientConfig)
+	leaderElectionConfig = rest.AddUserAgent(leaderElectionConfig, "kcp-leader-election")
+	// TODO(embik): is there a better way to access / generate the system:admin workspace url?
+	leaderElectionConfig.Host = fmt.Sprintf("%s/clusters/system:admin", leaderElectionConfig.Host)
+
+	hostname, err := os.Hostname()
```

Comment:
isn't that usually constant in containers? Do we need a flag?

Reply:
I looked at how kube-controller-manager does it and it also uses `os.Hostname`: https://github.com/kubernetes/kubernetes/blob/46835f8792dfb4a17345e592d1325bf63bc054e4/cmd/kube-controller-manager/app/controllermanager.go#L256

I guess it would be nice for this to work outside of container environments as well (e.g. on the local system).

Reply:
if it is a non-issue, all is good :-) just looked suspicious.